wallet/src/main.rs

```diff
@@ -80,18 +53,29 @@ pub fn parse_args(matches: &ArgMatches<'_>) -> Result<WalletConfig, Box<dyn erro
     let command = parse_command(&keypair.pubkey(), &matches)?;
 
     Ok(WalletConfig {
-        keypair,
         command,
         drone_host,
         drone_port,
-        host,
+        json_rpc_url,
+        keypair,
         rpc_client: None,
-        rpc_host,
-        rpc_port,
-        rpc_tls: matches.is_present("rpc_tls"),
     })
 }
 
+// Return an error if a url cannot be parsed.
+fn is_url(string: String) -> Result<(), String> {
```

Comment:
Should we collect clap validators somewhere general so they don't have to be copied across modules?
I'm a bit ambivalent about this, since there's not a great way to reuse other clap utility across modules.

Reply:
Yeah it seems not painful enough yet to support building another common thing to hold them.  Let's wait until this gets worse IMO